### PR TITLE
feat(Command): Remove spy command from AC and make it a module

### PR DESCRIPTION
**Pull request by Deku**
- Addresses the issue described in #1355 
- Adds some checks to the follow commands

PS: We can add other suggestions before merging this PR

==================================================

UPDATE: Since this command can bring trust issues between players and server owners, it will be made available as a module.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1549061780036243400.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1549061780036243400');
+
+DELETE FROM `trinity_string` WHERE `entry` = 5072;
+INSERT INTO `trinity_string` (`entry`, `content_default`) VALUES
+(5072, 'You can\'t spy yourself.');
```
Comment (BarbzYHOOL) [inline data/sql/updates/pending_db_world/rev_1549061780036243400.sql]: missing new line :D

File: src/server/game/Miscellaneous/Language.h [hunk 2]
```diff
@@ -1157,8 +1157,9 @@ enum TrinityStrings
     LANG_COMMAND_SPY_LIST                = 5069,
     LANG_COMMAND_SPY_WHISPERS_PLAYER     = 5070,
     LANG_COMMAND_SPY_TELLS_GROUP         = 5071,
+    LANG_COMMAND_SPY_SELF                = 5072,
 
-    // Room for more Trinity strings      5072-9999
+    // Room for more Trinity strings      5073-9999
```
Comment (BarbzYHOOL) [inline src/server/game/Miscellaneous/Language.h]: I know it's not you who wrote this but "Room for ..." sounds weird to me, is it correct english? If someone is not well versed into english, he might not understand what this room is for, no? xD

File: src/server/scripts/Commands/cs_spy.cpp [hunk 3]
```diff
@@ -128,28 +157,43 @@ class chat_spy_commandscript : public CommandScript
             return true;
         }
 
-        if(Player* target = ObjectAccessor::FindPlayer(sObjectMgr->GetPlayerGUIDByName(pTarget.c_str())))
+        Player* target = ObjectAccessor::FindPlayer(sObjectMgr->GetPlayerGUIDByName(pTarget.c_str()));
+
+        if (!target) {
```
Comment (BarbzYHOOL) [inline src/server/scripts/Commands/cs_spy.cpp]:
```suggestion
        if (!target)
        {
```

(for consistency)

File: src/server/game/Miscellaneous/Language.h (hunk 2 above)
Comment (Deku) [reply src/server/game/Miscellaneous/Language.h]: Don't blame me, that's in the original file, so I'm not gonna change it

Comment (BarbzYHOOL) [reply src/server/game/Miscellaneous/Language.h]: > I know it's not you who wrote this

**Comment by Deku** [comment]
After discussing with @BarbzYHOOL about possible trust issues with AzerothCore between players and server owners, we've decided to extract this command from the core and make it a module. That way anyone who wants it can implement it, taking into account the possible consequences it might bring.

**Comment by FrancescoBorzi** [comment]
@Deku sounds good, can you please clean up the code that we already have from the main repo?

File: data/sql/updates/pending_db_world/rev_1549462763029736000.sql [hunk 4]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1549462763029736000');
+
+DELETE FROM `trinity_string` WHERE `entry` BETWEEN 5062 AND 5071;
+DELETE FROM `command` WHERE `name` IN ('spy follow', 'spy unfollow', 'spy groupfollow', 'spy groupunfollow', 'spy clear', 'spy status');
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1549462763029736000.sql]: Missing end line

**Comment by Deku** [comment]
@FrancescoBorzi @BarbzYHOOL this is ready to be merged

**Comment by FrancescoBorzi** [comment]
@Deku there are conflicts

**Comment by BarbzYHOOL** [comment]
Should not have merged it before the module was done :'( (sorry wasn't there to say it before)

**Pull request merged**